### PR TITLE
Feat/get js avatar props

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Contributes to #446 

## What is the new behavior?

Adds Avatar JS API

### CODE REVIEW

**Comment by Rykus0** [comment]
Sandboxes
[SC](https://codesandbox.io/s/basic-avatar-css-in-js-sc-jb5imx?file=/src/App.tsx)

File: packages/headless-styles/src/components/Avatar/avatarJS.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  return {
+    avatar: {
+      a11yProps: props.avatar,
+      ...createJSProps(transformStyles(avatarStyles), avatarStyles),
+    },
+    image: {
+      a11yProps: props.image,
+      ...createJSProps(transformStyles(imageStyles), imageStyles),
+    },
+    iconOptions: props.iconOptions,
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/Avatar/avatarJS.ts:39]: Can kill this and just spread `...props` at the top of the `return`.

File: packages/headless-styles/tests/avatar/avatarJS.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const props = getJSAvatarProps()
+    expect(props.avatar.cssProps).toContain('border-radius: 50%')
+    expect(props.avatar.styles.borderRadius).toEqual('50%')
+    expect(props.image.cssProps).toContain('position: absolute')
+    expect(props.image.styles.position).toEqual('absolute')
+    expect(props.iconOptions).toEqual(defaultIconOptions)
+  })
+
+  test('should accept a kind type', () => {
+    const neutralProps = getJSAvatarProps({ kind: 'neutral' })
```
Comment (caseybaggz) [inline packages/headless-styles/tests/avatar/avatarJS.test.ts:20]: Don't forget to separate those tests! 😉

Comment (Rykus0) [reply packages/headless-styles/tests/avatar/avatarJS.test.ts:20]: I had written most of this before that feedback on the last one, and meant to come back to it and totally forgot 🤦

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**